legend-depot-core-services/src/main/java/org/finos/legend/depot/services/projects/DependenciesCache.java

```diff
@@ -65,22 +66,30 @@ private void initCache()
         }
     }
 
-    private Function3<String,String,String,StoreProjectVersionData> getProjectDataFromStore()
+    private Function3<String,String,String,StoreProjectVersionData> getProjectVersionDataFromStore()
     {
-        return (group, artifact, versionId) -> this.projectsVersionsStore.find(group, artifact,versionId).get();
+        return (group, artifact, versionId) ->
+        {
+            Optional<StoreProjectVersionData> projectVersion = this.projectsVersionsStore.find(group, artifact, versionId);
+            if (!projectVersion.isPresent())
+            {
+                throw new IllegalStateException(String.format("%s-%s-%s not found in store", group, artifact, versionId));
+            }
+            return this.projectsVersionsStore.find(group, artifact,versionId).get();
+        };
     }
 
-    private Function3<String,String,String,StoreProjectVersionData> getProjectDataFromProjectsMap(Map<String, StoreProjectVersionData> projectDataMap)
+    private Function3<String,String,String,StoreProjectVersionData> getProjectVersionDataFromProjectsMap(Map<String, StoreProjectVersionData> projectDataMap)
     {
         return (group, artifact, versionId) -> projectDataMap.get(group + artifact + versionId);
     }
 
-    private Set<ProjectVersion> calculateTransitiveDependencies(ProjectVersion projectVersion, Map<String, StoreProjectVersionData> projectDataMap)
+    private DependencyResult calculateTransitiveDependencies(ProjectVersion projectVersion, Map<String, StoreProjectVersionData> projectDataMap, DependencyStatus status)
     {
-        return getTransitiveDependencies(projectVersion, getProjectDataFromProjectsMap(projectDataMap));
+        return calculateTransitiveDependencies(projectVersion, getProjectVersionDataFromProjectsMap(projectDataMap), status);
     }
 
-    private Set<ProjectVersion> getTransitiveDependencies(ProjectVersion pv, Function3<String,String,String,StoreProjectVersionData> projectDataProvider)
+    private DependencyResult calculateTransitiveDependencies(ProjectVersion pv, Function3<String,String,String,StoreProjectVersionData> projectDataProvider, DependencyStatus status)
```

Comment:
do we use the status para, at all?

Reply:
mostly used to get the final status, we do need them, to figure whether the dep exists or not